src/math_ops.jl

```diff
@@ -21,6 +21,19 @@ julia> norm(A)
 @inline LinearAlgebra.norm(v::Vec) = sqrt(dot(v, v))
 @inline LinearAlgebra.norm(S::SecondOrderTensor) = sqrt(dcontract(S, S))
 
+@generated function LinearAlgebra.norm(S::Tensor{3,dim}) where {dim}
```

Comment:
I know this is copied from below, but I wonder how it compares to just delegating to StaticArrays.

Reply:
No major difference, but 5-10 % faster with Tensors' implementation on my system...

```julia
t = rand(Tensor{3,3});
snorm(t::Tensor) = norm(SVector(Tensors.get_data(t)))
tnorm(t::Tensor) = norm(Tensors.get_data(t))

julia> @btime norm($t);
  8.800 ns (0 allocations: 0 bytes) # Run 1
  9.500 ns (0 allocations: 0 bytes) # Run 2

julia> @btime snorm($t);
  10.000 ns (0 allocations: 0 bytes) # Run 1
  10.000 ns (0 allocations: 0 bytes) # Run 2

julia> @btime tnorm($t);
  87.137 ns (0 allocations: 0 bytes) # 🥱
```

Reply:
But if we really care about performance, and that would also reduce the amount of generated code...
```julia
julia> function _norm(t::NTuple{N,T}) where {N,T<:AbstractFloat}
           n = zero(T)
           @simd for v in t
               n += v^2
           end
           return sqrt(n)
       end
_norm (generic function with 1 method)

julia> _norm(t::Tensor) = _norm(Tensors.get_data(t))
_norm (generic function with 2 methods)

julia> @btime _norm($t3);
  4.000 ns (0 allocations: 0 bytes)
```